helios-services/src/main/java/com/spotify/helios/master/reaper/DeadAgentReaper.java

```diff
@@ -53,56 +54,63 @@
 
   public DeadAgentReaper(final MasterModel masterModel,
                          final long timeoutHours) {
-    this(masterModel, timeoutHours, SYSTEM_CLOCK);
+    this(masterModel, timeoutHours, SYSTEM_CLOCK, PERMITS_PER_SECOND,
+         new Random().nextInt(DELAY));
```

Comment:
Minor: `ThreadLocalRandom`?  Also this is logic in ctor, consider Effective Java item #1: create a static factory method.